src/modules/secrets/Elsa.Secrets/Models/Secret.cs

```diff
@@ -0,0 +1,12 @@
+using Elsa.Models;
+
+namespace Elsa.Secrets.Models
+{
+    public class Secret : Entity
+    {
+        public string Type { get; set; } = default!;
+        public string? Name { get; set; }
+        public string? DisplayName { get; set; }
+        public string PropertiesJson { get; set; }
```

Comment:
Suggestion: rename to `Data` and change its type to `object` or `JObject` or a dictionary. This way, serialization only needs to happen at the DB write/read step.